test/utils/delete_resources.go

```diff
@@ -52,14 +55,32 @@ func deleteResource(c clientset.Interface, kind schema.GroupKind, namespace, nam
 		return c.CoreV1().ConfigMaps(namespace).Delete(context.TODO(), name, options)
 	case api.Kind("Service"):
 		return c.CoreV1().Services(namespace).Delete(context.TODO(), name, options)
+	case kindCustomCRD:
+		if resourceClient != nil {
+			err1 := c.AppsV1().Deployments(namespace).Delete(context.TODO(), name, options)
+			err2 := resourceClient.Delete(context.TODO(), name, options)
+			if err1 != nil {
+				return err1
+			}
+			if err2 != nil {
+				return err2
+			}
+			return nil
+		} else {
+			return fmt.Errorf("Invalid resource client to delete object of kind: %v", kind)
+		}
 	default:
 		return fmt.Errorf("unsupported kind when deleting: %v", kind)
 	}
 }
 
 func DeleteResourceWithRetries(c clientset.Interface, kind schema.GroupKind, namespace, name string, options metav1.DeleteOptions) error {
+	return DeleteResourceWithRetriesithCustomResourceClient(c, nil, kind, namespace, name, options)
+}
+
+func DeleteResourceWithRetriesithCustomResourceClient(c clientset.Interface, resourceClient dynamic.ResourceInterface, kind schema.GroupKind, namespace, name string, options metav1.DeleteOptions) error {
```

Comment:
if we're making a new method, pass in a GroupVersionResource and dynamic client and support all types (can use the dynamic client to directly delete the specified GroupVersionResource rather than hard-coding a single new CRD type into this util method